### PR TITLE
Improve traces in distributor

### PR DESCRIPTION
**Pull request by pstibrany**
#### What this PR does

This PR adds extra log after incoming message in distributor has been unmarshalled, and adds tracing spans to distributor's forwarding code.

#### Checklist

- [na] Tests updated
- [na] Documentation added
- [na] `CHANGELOG.md` updated - the order of entries should be `[CHANGE]`, `[FEATURE]`, `[ENHANCEMENT]`, `[BUGFIX]`

### CODE REVIEW

File: pkg/distributor/forwarding/forwarding.go [hunk 1]
```diff
@@ -234,6 +237,8 @@ func (f *forwarder) Forward(ctx context.Context, endpoint string, rules validati
 	}
 
 	go func() {
+		defer spanlog.Finish()
```
Comment (56quarters) [inline pkg/distributor/forwarding/forwarding.go:240]: nit: there's no early returns in this function so this call doesn't really need to be deferred, does it?

File: pkg/distributor/forwarding/forwarding.go [hunk 2]
```diff
@@ -223,6 +224,8 @@ func (f *forwarder) Forward(ctx context.Context, endpoint string, rules validati
 		return in, errCh
 	}
 
+	spanlog, ctx := spanlogger.NewWithLogger(ctx, f.log, "forwarder.Forward")
```
Comment (56quarters) [inline pkg/distributor/forwarding/forwarding.go:227]: nit
```suggestion
	spanlog, ctx := spanlogger.NewWithLogger(ctx, f.log, "forwarder.Forward()")
```

File: pkg/util/http.go [hunk 3]
```diff
@@ -164,9 +164,17 @@ func ParseProtoReader(ctx context.Context, reader io.Reader, expectedSize, maxSi
 		err = proto.NewBuffer(body).Unmarshal(req)
 	}
 	if err != nil {
+		if sp != nil {
+			sp.LogFields(otlog.String("event", "util.ParseProtoRequest[unmarshal done]"), otlog.Error(err))
```
Comment (56quarters) [inline pkg/util/http.go:168]: nit: Apparently `otlog` has an `Event()` method that does exactly this.

**Review by 56quarters — APPROVED** [review]
LGTM, nothing but non-blocking superficial stuff.

File: pkg/distributor/forwarding/forwarding.go (hunk 2 above)
Comment (pstibrany) [reply pkg/distributor/forwarding/forwarding.go:229]: we actually don't use this style elsewhere. I will remove it from `request.do` span too.

File: pkg/util/http.go (hunk 3 above)
Comment (pstibrany) [reply pkg/util/http.go]: And now it's used in latest version of the PR. Thanks!

**Comment by pstibrany** [comment]
Thanks for review Nick!

**Pull request merged**